lucene/core/src/test/org/apache/lucene/index/TestKnnGraph.java

```diff
@@ -124,21 +126,84 @@ public void testMerge() throws Exception {
     }
   }
 
-  private void dumpGraph(KnnGraphValues values, int size) throws IOException {
+  /**
+   * Verify that we get the *same* graph by indexing one segment as we do by indexing two segments
+   * and merging.
+   */
+  public void testMergeProducesSameGraph() throws Exception {
```

Comment:
Nice test :)

Reply:
thanks, yes this flushed out the two problems I saw, so I'm pretty confident they are fixed now, after running this a few 100 times. I had also wanted to add a test asserting that KNN search precision remains above some threshold, but sadly with random vectors, it would always eventually fail, even though mostly it would succeed, so not a very useful unit test and I removed it. Probably we can add something to luceneutil